cmd/kube-apiserver/app/preflight/checks.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"time"
+
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+)
+
+const retryLimit = 60
+const retryInterval = 1 * time.Second
+
+type clock interface {
+	Sleep(time.Duration)
+	After(time.Duration) <-chan time.Time
+}
+
+type connectionTimer struct{}
```

Comment:
no need for this.  Just use the util/clock.  apiservers already depend on client-go.